Restrict the propagation of invalid txns

ptschip opened the pull request.

ptschip committed:
Be more restrictive in what txns are considered valid

There were some txns that were getting propagated that were not
proper Bitcoin Cash transactions.

ptschip committed:
Fix signrawtransaction which was causing many regression test failures

ptschip committed:
Fix two regression tests

Need to mine some blocks before sending transactions

ptschip committed:
use default nHashType SIGHASH_FORKID when signing

When BITCOIN_CASH is defined then make sure if no nHashType is used
then we default to SIGHASH_ALL|SIGHASH_FORKID

gandrewstone commented on qa/rpc-tests/listtransactions.py:
```diff
@@ -20,12 +20,12 @@ def txFromHex(hexstring):
 
     def setup_nodes(self):
-        #This test requires mocktime
-        enable_mocktime()
```
did you delete the cache directory?  mocktime (and this test) should work since I changed the mock time to be post fork.  But you need to delete your node cache to force it to be regenerated with the new mocktime.

ptschip committed:
Renable the use of mocktime

These tests were working fine before.  All that was needed was to
delete the cache.

ptschip committed:
Temporarily do not ban BitcoinCash BU nodes

We need to do this to be backwardly compatible with nodes that
are falsely forwarding bad txns.  We don't want to end up banning
them.

On the next release we can take this out.
